### PR TITLE
Added NDCMatchFilter and MDCMatchFilter to 2.0.x

### PR DESCRIPTION
**Pull request by francky06l**
Added NDCMatchFilter and MDCMatchFilter to 2.0.x (master requires GCC 7  for c++17)

### CODE REVIEW

File: include/log4cplus/spi/filter.h [hunk 1]
```diff
@@ -303,6 +303,57 @@ namespace log4cplus {
             Function function;
         };
 
+        class LOG4CPLUS_EXPORT NDCMatchFilter : public Filter
```
Comment (wilx) [inline include/log4cplus/spi/filter.h]: Could you please add some Doxygen documentation describing the filter?

File: include/log4cplus/spi/filter.h [hunk 2]
```diff
@@ -303,6 +303,57 @@ namespace log4cplus {
...
+
+              // Data
+                /** Do we return ACCEPT when a match occurs. Default is <code>true</code>. */
+                bool acceptOnMatch;
+                /** return NEUTRAL if NDC::get() is empty or ndcToMatch is empty. Default is <code>true</code>. */
+                bool neutralOnEmpty;
+                log4cplus::tstring ndcToMatch;
+        };
+
+        class LOG4CPLUS_EXPORT MDCMatchFilter : public Filter
```
Comment (wilx) [inline include/log4cplus/spi/filter.h]: Same here. Could you please add some Doxygen documentation describing the filter?

File: src/filter.cxx [hunk 3]
```diff
@@ -273,6 +273,88 @@ FunctionFilter::decide(const InternalLoggingEvent& event) const
...
+
+    if(neutralOnEmpty && mdcStr.empty())
+        return NEUTRAL;
+
+    if(mdcStr.compare(mdcValueToMatch) == 0)
+        return (acceptOnMatch ? ACCEPT : DENY);
+
+    return (acceptOnMatch ? DENY : ACCEPT);
+}
+
```
Comment (wilx) [inline src/filter.cxx]: It would also be nice to add some unit tests.

Comment (francky06l) [reply src/filter.cxx]: This done in the second commit.
Thanks

**Pull request merged**